### DISCUSSION TITLE
Grouping notifications by topic, and pagination

### DISCUSSION THREAD

**Discussion by scpatt** [magicbell/community#99] [Questions] [answered]
Hello! I've been exploring the Magicbell API recently as I'm considering using it in a project of mine. 

I like the idea of associating notifications with uniquely identifiable topics, but it seems (at least from my brief testing of the REST API) that it isn't possible to have these notifications grouped by the API. 

My expectation when assigning a notification a topic, is that all notifications for that topic would be returned as a single item in the response pre-grouped, or at least a query parameter that would enable this. My goal is to have notifications which would be rendered similar to social media websites where you would see something like "Person A and 3 others liked your comment".

Without this behaviour pagination becomes a problem, because if I group the notifications on the client I haven't necessarily fetched a full page of notifications.

Am I misunderstanding how topics are supposed to work, or is this behaviour intended?

Thanks :)

**Comment by unamashana** [comment, marked as answer]
Thanks for writing in @Astronought! 

You can fetch notifications for a topic by adding the query param `topics`, which is an array of topic slugs (`[]string`). The results are also paginated.

We are also going to improve topics significantly for the new Topical Inbox. We talked about this in our customer event last week - https://youtu.be/d71sEE94y7Q
